File: TestProject1/Module/UnitTest_MdlFile.cs
using CmnClsLib.Module;
using Xunit;
using Assert = Xunit.Assert;

namespace TestProject1.Module
{
    public class UnitTest_MdlFile
    {

        // ls ${Env:USERPROFILE}\AppData\Local\Temp\UnitTest
        private string _tempdir = Path.Combine(System.IO.Path.GetTempPath(), @"UnitTest", @"CmnClsLib", @"MdlFile");

        // --------------------------------------------------------------------
        // GetDirectoryPath()
        // --------------------------------------------------------------------
        [Fact]
        public void GetDirectoryPath_引数で指定したファイルのディレクトリパスを返却すること()
        {
            string target = Path.Combine(_tempdir, @"test.txt");
            string expected = _tempdir;
            Assert.Equal(expected, MdlFile.GetDirectoryPath(target));
        }

        // --------------------------------------------------------------------
        // GetFileNameWithoutExtension()
        // --------------------------------------------------------------------
        [Fact]
        public void GetFileNameWithoutExtension_引数で指定したファイルの拡張子無しファイル名を返却すること()
        {
            string target = Path.Combine(_tempdir, @"test.txt");
            string expected = @"test";
            Assert.Equal(expected, MdlFile.GetFileNameWithoutExtension(target));
        }

        // --------------------------------------------------------------------
        // GetFileName()
        // --------------------------------------------------------------------
        [Fact]
        public void GetFileName_引数で指定したファイルの拡張子有しファイル名を返却すること()
        {
            string target = Path.Combine(_tempdir, @"test.txt");
            string expected = @"test.txt";
            Assert.Equal(expected, MdlFile.GetFileName(target));
        }

        // --------------------------------------------------------------------
        // GetFileExtension()
        // --------------------------------------------------------------------
        [Fact]
        public void GetFileExtension_引数で指定したファイルの拡張子を返却すること()
        {
            string target = Path.Combine(_tempdir, @"test.txt");
            string expected = @"txt";
            Assert.Equal(expected, MdlFile.GetFileExtension(target));
        }

        // --------------------------------------------------------------------
        // CreateDirectory()
        // --------------------------------------------------------------------
        [Fact]
        public void CreateDirectory_引数が空文字列の場合は定数値NG_MKDIR_WRONG_ARGを返却すること()
        {
            string target = @"";
            int expected = MdlFile.NG_MKDIR_WRONG_ARG;
            int actual = MdlFile.CreateDirectory(target);
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void CreateDirectory_引数に指定したディレクトリの作成に成功した場合は定数値OK_MKDIR_CREATE_0を返却すること()
        {
            string target = Path.Combine(_tempdir, @"CreateDirectory\002");
            int expected = MdlFile.OK_MKDIR_CREATE;
            int actual = -1;
            bool isOk = false;
            isOk = MdlFile.DeleteRecursively(target);
            actual = MdlFile.CreateDirectory(target);
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void CreateDirectory_引数に既に存在するディレクトリを指定した場合は定数値OK_MKDIR_ALREADY_EXIT_1を返却すること()
        {
            string target = Path.Combine(_tempdir, @"CreateDirectory\003");
            int actual = -1;
            int expected = MdlFile.OK_MKDIR_ALREADY_EXIST;
            actual = MdlFile.CreateDirectory(target);
            actual = MdlFile.CreateDirectory(target);
            Assert.Equal(expected, actual);

        }

        [Fact]
        public void CreateDirectory_引数に既に存在するファイルを指定した場合は定数値NG_MKDIR_FILE_EXIST_13を返却すること()
        {
            bool isOk = false;
            string target = Path.Combine(_tempdir, @"CreateDirectory\AlreadyExist.txt");
            int actual = -1;
            int expected = MdlFile.NG_MKDIR_FILE_EXIST;
            isOk = MdlFile.DeleteRecursively(target);
            actual = MdlFile.CreateEmptyFile(target);
            actual = MdlFile.CreateDirectory(target);
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void CreateDirectory_引数に指定したディレクトリの作成に失敗した場合は定数値NG_MKDIR_11を返却すること()
        {
            int actual = -1;
            string target = @"A:\NO_DRIVE";
            int expected = MdlFile.NG_MKDIR;
            actual = MdlFile.CreateDirectory(target);
            Assert.Equal(expected, actual);
        }

        // --------------------------------------------------------------------
        // CreateEmptyFile()
        // --------------------------------------------------------------------
        [Fact]
        public void Touch_引数で指定されたファイルの作成に成功した場合OK_TOUCH_CREATE_0を返却すること()
        {
            int actual = -1;
            bool isOk = false;
            string target = Path.Combine(_tempdir, @"CreateEmptyFile\ok.txt");
            int expected = MdlFile.OK_TOUCH_CREATE;
            isOk = MdlFile.DeleteRecursively(target);
            actual = MdlFile.CreateEmptyFile(target);
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void Touch_引数で指定されたファイルが既に存在する場合OK_TOUCH_ALREADY_EXIST_1を返却すること()
        {
            int actual = -1;
            bool isOk = false;
            string target = Path.Combine(_tempdir, @"CreateEmptyFile\alreadyexist.txt");
            int expected = MdlFile.OK_TOUCH_ALREADY_EXIST;
            isOk = MdlFile.DeleteRecursively(target);
            actual = MdlFile.CreateEmptyFile(target);
            actual = MdlFile.CreateEmptyFile(target);
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void Touch_引数で指定されたファイルがディレクトリとして既に存在する場合NG_TOUCH_DIR_EXIST_13を返却すること()
        {
            int actual = -1;
            bool isOk = false;
            string target = Path.Combine(_tempdir, @"CreateEmptyFile\AlreadyExsitFolder");
            int expected = MdlFile.NG_TOUCH_DIR_EXIST;
            isOk = MdlFile.DeleteRecursively(target);
            actual = MdlFile.CreateDirectory(target);
            actual = MdlFile.CreateEmptyFile(target);
            Assert.Equal(expected, actual);
        }

        // --------------------------------------------------------------------
        // GetPathType()
        // --------------------------------------------------------------------
        [Fact]
        public void GetPathType_引数が存在するディレクトリの場合は定数値PATH_IS_DIRECTORY_1を返却すること()
        {
            string target = Path.Combine(_tempdir, @"GetPathType\001");
            int actual = -1;
            bool isOk = false;
            int expected = MdlFile.PATH_IS_DIRECTORY;
            isOk = MdlFile.DeleteRecursively(target);
            actual = MdlFile.CreateDirectory(target);
            actual = MdlFile.GetPathType(target);
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void GetPathType_引数が存在するファイルの場合は定数値PATH_IS_FILEを返却すること()
        {
            int actual = -1;
            bool isOk = false;
            string target = Path.Combine(_tempdir, @"GetPathType\002.txt");
            int expected = MdlFile.PATH_IS_FILE;
            isOk = MdlFile.DeleteRecursively(target);
            actual = MdlFile.CreateEmptyFile(target);
            actual = MdlFile.GetPathType(target);
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void GetPathType_引数が空文字列の場合は定数値PATH_IS_NULLを返却すること()
        {
            int actual = -1;
            string target = @"";
            int expected = MdlFile.PATH_IS_NULL;
            actual = MdlFile.GetPathType(target);
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void GetPathType_引数が存在しないパスの場合は定数値PATH_NOT_FOUNDを返却すること()
        {
            int actual = -1;
            string target = @"C:\NOT_EXIT";
            int expected = MdlFile.PATH_NOT_FOUND;
            actual = MdlFile.GetPathType(target);
            Assert.Equal(expected, actual);
        }

        // --------------------------------------------------------------------
        // PathExists()
        // --------------------------------------------------------------------
        [Fact]
        public void PathExists_引数が存在するディレクトリの場合はtrueを返却すること()
        {
            string target = Path.Combine(_tempdir, @"PathExists\001");
            MdlFile.CreateDirectory(target);
            bool actual = MdlFile.PathExists(target);
            Assert.True(actual);
        }

        [Fact]
        public void PathExists_引数が存在するファイルの場合はtrueを返却すること()
        {
            string target = Path.Combine(_tempdir, @"PathExists\002.txt");
            MdlFile.CreateEmptyFile(target);
            bool actual = MdlFile.PathExists(target);
            Assert.True(actual);
        }

        [Fact]
        public void PathExists_引数が空文字列の場合はfalseを返却すること()
        {
            string target = @"";
            bool actual = MdlFile.PathExists(target);
            Assert.False(actual);
        }

        [Fact]
        public void PathExists_引数が存在しないパスの場合はfalseを返却すること()
        {
            string target = @"C:\NOT_EXIT";
            bool actual = MdlFile.PathExists(target);
            Assert.False(actual);
        }

        // --------------------------------------------------------------------
        // RemoveTrailingPathSeparator()
        // --------------------------------------------------------------------
        [Fact]
        public void RemoveTrailingPathSeparator_引数の文字列の最後に円マークが指定された場合は除去して返却すること()
        {
            string target = @"C:\DIR1\DIR2\";
            string expected = @"C:\DIR1\DIR2";
            string actual = MdlFile.RemoveTrailingPathSeparator(target);
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void RemoveTrailingPathSeparator_引数の文字列の最後にスラッシュが指定された場合は除去して返却すること()
        {
            string target = @"C:/DIR1/DIR2/";
            string expected = @"C:\DIR1\DIR2";
            string actual = MdlFile.RemoveTrailingPathSeparator(target);
            Assert.Equal(expected, actual);
        }

        // --------------------------------------------------------------------
        // GetAbsolutePath()
        // --------------------------------------------------------------------
        [Fact]
        public void GetAbsolutePath_引数が相対パスの場合に絶対パスを返却すること()
        {
            // 先頭が一致すればドライブ文字が含まれる？
            string target = @"..\";
            string expected = System.IO.Directory.GetCurrentDirectory().Substring(0, 3);
            string actual = MdlFile.GetAbsolutePath(target).Substring(0, 3);
            Assert.Equal(expected, actual);
        }

        // --------------------------------------------------------------------
        // SanitizeFileName()
        // --------------------------------------------------------------------
        [Fact]
        public void SanitizeFileName_引数に空白が含まれる場合はアンダースコアに置換して返却すること()
        {
            string expected = @"a_b";
            string actual = MdlFile.SanitizeFileName("a b");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数に円マークが含まれる場合はアンダースコアに置換して返却すること()
        {
            string expected = @"a_b";
            string actual = MdlFile.SanitizeFileName(@"a\b");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数にスラッシュが含まれる場合はアンダースコアに置換して返却すること()
        {
            string expected = @"a_b";
            string actual = MdlFile.SanitizeFileName(@"a/b");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数にコロンが含まれる場合は除去して返却すること()
        {
            string expected = @"abc";
            string actual = MdlFile.SanitizeFileName(@"a:b:c");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数にセミコロンが含まれる場合は除去して返却すること()
        {
            string expected = @"abc";
            string actual = MdlFile.SanitizeFileName(@"a;b;c");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数にパイプが含まれる場合は除去して返却すること()
        {
            string expected = @"abc";
            string actual = MdlFile.SanitizeFileName(@"a|b|c");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数にカンマが含まれる場合は除去して返却すること()
        {
            string expected = @"abc";
            string actual = MdlFile.SanitizeFileName(@"a,b,c");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数にアスタリスクが含まれる場合は除去して返却すること()
        {
            string expected = @"abc";
            string actual = MdlFile.SanitizeFileName(@"a*b*c");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数にクエスチョンマークが含まれる場合は除去して返却すること()
        {
            string expected = @"abc";
            string actual = MdlFile.SanitizeFileName(@"a?b?c");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数に小なりが含まれる場合は除去して返却すること()
        {
            string expected = @"abc";
            string actual = MdlFile.SanitizeFileName(@"a<b<c");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数に大なりが含まれる場合は除去して返却すること()
        {
            string expected = @"abc";
            string actual = MdlFile.SanitizeFileName(@"a>b>c");
            Assert.Equal(expected, actual);
        }

        [Fact]
        public void SanitizeFileName_引数にダブルクォーテーションが含まれる場合は除去して返却すること()
        {
            string expected = @"abc";
            string actual = MdlFile.SanitizeFileName("a\"b\"c");
            Assert.Equal(expected, actual);
        }

        // --------------------------------------------------------------------
        // WriteFile() && ReadFile
        // --------------------------------------------------------------------
        [Fact]
        public void WriteFileAndReadFile_ファイル書込とファイル読込ログファイルができること()
        {
            // 変数初期化
            string logDir = Path.Combine(_tempdir, @"WriteFile_ReadFile");
            string logPath = Path.Combine(logDir, @"test.txt");
            string expected = "TESTTESTTEST";
            // 事前作業
            MdlFile.CreateDirectory(logDir);
            if (MdlFile.GetPathType(logPath) == MdlFile.PATH_IS_FILE)
            {
                System.IO.File.Delete(logPath);
            }
            // 出力
            MdlFile.WriteFile(logPath, expected);
            // テスト
            string actual = MdlFile.ReadFile(logPath, -1);
            Assert.Equal(expected, actual);
            // 事後作業
            if (MdlFile.GetPathType(logPath) == MdlFile.PATH_IS_FILE)
            {
                System.IO.File.Delete(logPath);
            }
        }
        // --------------------------------------------------------------------
        // DeleteRecursively()
        // --------------------------------------------------------------------
        [Fact]
        public void rm_rf_引数で指定したディレクトリの削除に成功した場合はtrueを返却すること()
        {
            string target = @"C:\Tool\UnitTest\MdlFile";
            Assert.True(MdlFile.DeleteRecursively(target));
        }
    }
}

File: src/test/java/tool/cmnclslib/mdl/UnitTest_MdlFile.java
package tool.cmnclslib.mdl;

import java.io.File;
import org.junit.Test;
import static org.junit.Assert.*;

/**
 * MdlFile の単体テストクラスです。
 */
public class UnitTest_MdlFile {

    private String tempDir = new File(new File(new File(System.getProperty("java.io.tmpdir"), "UnitTest"), "CmnClsLib"), "MdlFile").getPath();

    @Test
    public void getDirectoryPath_引数で指定したファイルのディレクトリパスを返却すること() {
        String target = tempDir + File.separator + "test.txt";
        String expected = tempDir;
        assertEquals(expected, MdlFile.getDirectoryPath(target));
    }

    @Test
    public void getFileNameWithoutExtension_引数で指定したファイルの拡張子無しファイル名を返却すること() {
        String target = tempDir + File.separator + "test.txt";
        String expected = "test";
        assertEquals(expected, MdlFile.getFileNameWithoutExtension(target));
    }

    @Test
    public void getFileName_引数で指定したファイルの拡張子有しファイル名を返却すること() {
        String target = tempDir + File.separator + "test.txt";
        String expected = "test.txt";
        assertEquals(expected, MdlFile.getFileName(target));
    }

    @Test
    public void getFileExtension_引数で指定したファイルの拡張子を返却すること() {
        String target = tempDir + File.separator + "test.txt";
        String expected = "txt";
        assertEquals(expected, MdlFile.getFileExtension(target));
    }

    @Test
    public void createDirectory_引数が空文字列の場合は定数値NG_MKDIR_WRONG_ARGを返却すること() {
        String target = "";
        int expected = MdlFile.NG_MKDIR_WRONG_ARG;
        int actual = MdlFile.createDirectory(target);
        assertEquals(expected, actual);
    }

    @Test
    public void createDirectory_引数に指定したディレクトリの作成に成功した場合は定数値OK_MKDIR_CREATE_0を返却すること() {
        String target = tempDir + File.separator + "CreateDirectory" + File.separator + "002";
        int expected = MdlFile.OK_MKDIR_CREATE;
        MdlFile.deleteRecursively(target);
        int actual = MdlFile.createDirectory(target);
        assertEquals(expected, actual);
    }

    @Test
    public void createDirectory_引数に既に存在するディレクトリを指定した場合は定数値OK_MKDIR_ALREADY_EXIT_1を返却すること() {
        String target = tempDir + File.separator + "CreateDirectory" + File.separator + "003";
        int expected = MdlFile.OK_MKDIR_ALREADY_EXIST;
        MdlFile.createDirectory(target);
        int actual = MdlFile.createDirectory(target);
        assertEquals(expected, actual);
    }

    @Test
    public void createDirectory_引数に既に存在するファイルを指定した場合は定数値NG_MKDIR_FILE_EXIST_13を返却すること() {
        String target = tempDir + File.separator + "CreateDirectory" + File.separator + "AlreadyExist.txt";
        int expected = MdlFile.NG_MKDIR_FILE_EXIST;
        MdlFile.deleteRecursively(target);
        MdlFile.createEmptyFile(target);
        int actual = MdlFile.createDirectory(target);
        assertEquals(expected, actual);
    }

    @Test
    public void createEmptyFile_引数で指定されたファイルの作成に成功した場合OK_TOUCH_CREATE_0を返却すること() {
        String target = tempDir + File.separator + "CreateEmptyFile" + File.separator + "ok.txt";
        int expected = MdlFile.OK_TOUCH_CREATE;
        MdlFile.deleteRecursively(target);
        int actual = MdlFile.createEmptyFile(target);
        assertEquals(expected, actual);
    }

    @Test
    public void createEmptyFile_引数で指定されたファイルが既に存在する場合OK_TOUCH_ALREADY_EXIST_1を返却すること() {
        String target = tempDir + File.separator + "CreateEmptyFile" + File.separator + "alreadyexist.txt";
        int expected = MdlFile.OK_TOUCH_ALREADY_EXIST;
        MdlFile.deleteRecursively(target);
        MdlFile.createEmptyFile(target);
        int actual = MdlFile.createEmptyFile(target);
        assertEquals(expected, actual);
    }

    @Test
    public void createEmptyFile_引数で指定されたファイルがディレクトリとして既に存在する場合NG_TOUCH_DIR_EXIST_13を返却すること() {
        String target = tempDir + File.separator + "CreateEmptyFile" + File.separator + "AlreadyExsitFolder";
        int expected = MdlFile.NG_TOUCH_DIR_EXIST;
        MdlFile.deleteRecursively(target);
        MdlFile.createDirectory(target);
        int actual = MdlFile.createEmptyFile(target);
        assertEquals(expected, actual);
    }

    @Test
    public void getPathType_引数が存在するディレクトリの場合は定数値PATH_IS_DIRECTORY_1を返却すること() {
        String target = tempDir + File.separator + "GetPathType" + File.separator + "001";
        int expected = MdlFile.PATH_IS_DIRECTORY;
        MdlFile.deleteRecursively(target);
        MdlFile.createDirectory(target);
        int actual = MdlFile.getPathType(target);
        assertEquals(expected, actual);
    }

    @Test
    public void getPathType_引数が存在するファイルの場合は定数値PATH_IS_FILEを返却すること() {
        String target = tempDir + File.separator + "GetPathType" + File.separator + "002.txt";
        int expected = MdlFile.PATH_IS_FILE;
        MdlFile.deleteRecursively(target);
        MdlFile.createEmptyFile(target);
        int actual = MdlFile.getPathType(target);
        assertEquals(expected, actual);
    }

    @Test
    public void getPathType_引数が空文字列の場合は定数値PATH_IS_NULLを返却すること() {
        int expected = MdlFile.PATH_IS_NULL;
        int actual = MdlFile.getPathType("");
        assertEquals(expected, actual);
    }

    @Test
    public void getPathType_引数が存在しないパスの場合は定数値PATH_NOT_FOUNDを返却すること() {
        int expected = MdlFile.PATH_NOT_FOUND;
        int actual = MdlFile.getPathType("C:\\NOT_EXIST_PATH_XYZ_12345");
        assertEquals(expected, actual);
    }

    @Test
    public void pathExists_引数が存在するディレクトリの場合はtrueを返却すること() {
        String target = tempDir + File.separator + "PathExists" + File.separator + "001";
        MdlFile.createDirectory(target);
        assertTrue(MdlFile.pathExists(target));
    }

    @Test
    public void pathExists_引数が存在するファイルの場合はtrueを返却すること() {
        String target = tempDir + File.separator + "PathExists" + File.separator + "002.txt";
        MdlFile.createEmptyFile(target);
        assertTrue(MdlFile.pathExists(target));
    }

    @Test
    public void pathExists_引数が空文字列の場合はfalseを返却すること() {
        assertFalse(MdlFile.pathExists(""));
    }

    @Test
    public void pathExists_引数が存在しないパスの場合はfalseを返却すること() {
        assertFalse(MdlFile.pathExists("C:\\NOT_EXIST_PATH_XYZ_12345"));
    }

    @Test
    public void removeTrailingPathSeparator_引数の文字列の最後に円マークが指定された場合は除去して返却すること() {
        String target = "C:\\DIR1\\DIR2\\";
        String expected = "C:\\DIR1\\DIR2";
        assertEquals(expected, MdlFile.removeTrailingPathSeparator(target));
    }

    @Test
    public void sanitizeFileName_引数に空白が含まれる場合はアンダースコアに置換して返却すること() {
        assertEquals("a_b", MdlFile.sanitizeFileName("a b"));
    }

    @Test
    public void sanitizeFileName_引数に円マークが含まれる場合はアンダースコアに置換して返却すること() {
        assertEquals("a_b", MdlFile.sanitizeFileName("a\\b"));
    }

    @Test
    public void sanitizeFileName_引数にスラッシュが含まれる場合はアンダースコアに置換して返却すること() {
        assertEquals("a_b", MdlFile.sanitizeFileName("a/b"));
    }

    @Test
    public void sanitizeFileName_引数にコロンが含まれる場合は除去して返却すること() {
        assertEquals("abc", MdlFile.sanitizeFileName("a:b:c"));
    }

    @Test
    public void sanitizeFileName_引数にセミコロンが含まれる場合は除去して返却すること() {
        assertEquals("abc", MdlFile.sanitizeFileName("a;b;c"));
    }

    @Test
    public void sanitizeFileName_引数にパイプが含まれる場合は除去して返却すること() {
        assertEquals("abc", MdlFile.sanitizeFileName("a|b|c"));
    }

    @Test
    public void sanitizeFileName_引数にカンマが含まれる場合は除去して返却すること() {
        assertEquals("abc", MdlFile.sanitizeFileName("a,b,c"));
    }

    @Test
    public void sanitizeFileName_引数にアスタリスクが含まれる場合は除去して返却すること() {
        assertEquals("abc", MdlFile.sanitizeFileName("a*b*c"));
    }

    @Test
    public void sanitizeFileName_引数にクエスチョンマークが含まれる場合は除去して返却すること() {
        assertEquals("abc", MdlFile.sanitizeFileName("a?b?c"));
    }

    @Test
    public void sanitizeFileName_引数に小なりが含まれる場合は除去して返却すること() {
        assertEquals("abc", MdlFile.sanitizeFileName("a<b<c"));
    }

    @Test
    public void sanitizeFileName_引数に大なりが含まれる場合は除去して返却すること() {
        assertEquals("abc", MdlFile.sanitizeFileName("a>b>c"));
    }

    @Test
    public void sanitizeFileName_引数にダブルクォーテーションが含まれる場合は除去して返却すること() {
        assertEquals("abc", MdlFile.sanitizeFileName("a\"b\"c"));
    }

    @Test
    public void writeFileAndReadFile_ファイル書込とファイル読込ができること() {
        String logDir = tempDir + File.separator + "WriteFile_ReadFile";
        String logPath = logDir + File.separator + "test.txt";
        String expected = "TESTTESTTEST";

        MdlFile.createDirectory(logDir);
        new File(logPath).delete();

        MdlFile.writeFile(logPath, expected);
        String actual = MdlFile.readFile(logPath, -1);
        assertEquals(expected, actual);

        new File(logPath).delete();
    }

    @Test
    public void 新設メソッド_getBaseName_trimPathSeparator_readFileToMap等が正しく動作すること() {
        assertEquals("test", MdlFile.getBaseName("C:\\dir\\test.txt"));
        assertEquals("C:\\dir", MdlFile.trimPathSeparator("C:\\dir\\"));
        assertTrue(MdlFile.isValidDirDateTime(tempDir, false, null, false, null));
        assertEquals("cmd C:\\file.txt", MdlFile.replacePathForCmd("cmd _PATH_", "C:\\file.txt", "file.txt", false, 0));
        assertNotNull(MdlFile.getSortedDirsInfo(tempDir, "*", false, MdlFile.SORT_BY_NAME, true));
    }
}
